### ISSUE TITLE
await _player.seek never completes on iOS

### ISSUE THREAD

**Issue by snaeji** [ryanheise/just_audio#206]
**Which API doesn't behave as documented, and how does it misbehave?**

Calling `await _player.seek(Duration.zero, index:1)` never completes

**Minimal reproduction project**
https://github.com/snaeji/just_audio/tree/issue-branch
https://github.com/snaeji/just_audio/blob/issue-branch/just_audio/example/lib/main.dart

**To Reproduce (i.e. user steps, not code)**
1. Run example with `_init3()` - `_init4()` function

There is a` //TODO:` in the code so it should be easy to find

**Expected behavior**
The function should complete

**Smartphone (please complete the following information):**
 - Device: iPhone SE (Second Generation)
 - OS: iOS 14.0.1

**Flutter SDK version**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.22.1, on Mac OS X 10.15.7 19H2, locale en-GB)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
[!] Android Studio (version 4.1)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] IntelliJ IDEA Community Edition (version 2018.2.4)
[✓] VS Code (version 1.49.3)
[✓] Connected device (1 available)
    ! Error: Auglysingadeild4thgen is not connected. Xcode will continue when Auglysingadeild4thgen is connected. (code -13)
    ! Error: Living Room is not connected. Xcode will continue when Living Room is connected. (code -13)
    ! Error: Living Room is not connected. Xcode will continue when Living Room is connected. (code -13)

! Doctor found issues in 1 category.
```

**Additional context**
May be closely related to https://github.com/ryanheise/just_audio/issues/207

**Comment by ryanheise** [comment]
Thanks for the example. When I tried running init 4, it took a few minutes to actually finish loading the audio source. Not sure if it's the same on a real device, but this in itself is not desirable behaviour so I'll look into that. I can confirm that seek never completes after loading eventually completes.

Init 3 actually gave me errors loading some items. I wonder if the server doesn't like us loading the same URL multiple times.

Is it intentional that you didn't await the seek in init 3?

**Comment by snaeji** [comment]
I think the long loading time is just likely to be an emulator problem. Everything loads right away on a physical device.
And I havent noticed any errors when running any of the inits on a physical device.

Not awaiting seek was intentional, my thinking went a little like this...

Init2: Show that position never seeks when index is changed on a large playlist
Init3: Show that position never seeks when index is changed on a small playlist
Init4: Show that seek never completes.

The reason for not awaiting seek in 2 - 3 is so we can start playing and hear the music play from start, but with the seek position indicated in the player UI.

**Comment by colehorvitz** [comment]
Any update on this? I wish for a track to play once seeking to Duration.zero on a new queue item is completed, but unfortunately, the future never resolves.

**Comment by ryanheise** [comment]
Thank you for bringing this back to my attention! I had a lot of items on the backlog and unfortunately lost track of this one. I have changed its status to "fixing" which means that I will take a look after finishing what I'm currently working on, which includes null-safety.

From memory, when I first investigated this issue, I was also at a loss. But taking a look at it again with fresh eyes, I now have some theories I would like to test.

**Comment by colehorvitz** [comment]
No problem, and thanks for all you do to maintain this great plugin!

**Comment by ryanheise** [comment]
I have just tested this, and it seems to be fixed along with #207 . Although perhaps if you have a long enough list, the loading will be so long that will seem to take forever? If it is an issue with the length of the playlist, I would be interested to know if you could find the threshold of how many items or minutes of audio it takes to cause a problem. This may be related to #294 .

Let me know if it's the same problem as #294 and I'll close this issue.

**Comment by colehorvitz** [comment]
Hi Ryan, I have just tested on the newest version and am unfortunately still experiencing the bug even on a very small dataset. I have reproduced the issue here: 

https://github.com/colehorvitz/just_audio
 
I have modified the callbacks on the seek to next and seek to previous buttons to demonstrate the behavior. When the seek prev/next buttons are pressed, "seeking to next/prev..." is printed to the console.  player.seekToNext() is then awaited, and when it resolves "done" should be printed onto the console. However, the future does not resolve and "done" never prints.

**Comment by ryanheise** [comment]
Thanks for the reproduction case. I have confirmed it and will take a look now.

**Comment by ryanheise** [comment]
Can you try the latest commit on `master`?

**Comment by colehorvitz** [comment]
That did the trick, it's working now. Thanks!

**Comment by ryanheise** [comment]
Great! This is now released in 0.6.12. Thanks again for providing the reproduction case.
